### PR TITLE
Create embed enhancement

### PR DESCRIPTION
**Pull request by Michael55555**
**Please describe the changes this PR makes and why it should be merged:**
The changes in this PR include an enhancement to the "Create Embed Message" mod, that you can already add the embed description in this action, instead of having to add a separate action to it.

**Status**

- [ ] Code changes have been tested against the Discord API and the discord.js wrapper, or there are no code changes
- [ ] Documentation has been added/modified, or there is nothing to change (docs/mods.json)

**Semantic versioning classification:**

- [x] This PR changes DBM's interface (methods or parameters added to default methods)
- [ ] This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

**Review by almeidx — CHANGES_REQUESTED** [review]
Unnecessary. This would defeat the purpose of the `Set Embed Description` action.

**Comment by Michael55555** [comment]
> Unnecessary. This would defeat the purpose of the `Set Embed Description` action.

Kind of, but most people would be happy is their action lists would be a bit shorter. Sometimes they can get really long. I am sure that people would use `Set Embed Description` in case they have to do it separately for their specific use case.

File: actions/create_embed_message.js [hunk 1]
```diff
@@ -9,124 +9,189 @@ module.exports = {
   variableStorage (data, varType) {
     const type = parseInt(data.storage)
     if (type !== varType) return
-    return ([data.varName, 'Embed Object'])
+    return [data.varName, 'Embed Object']
   },
 
-  fields: ['title', 'author', 'color', 'url', 'authorIcon', 'authorUrl', 'imageUrl', 'thumbUrl', 'timestamp', 'debug', 'timestamp1', 'timestamp2', 'text', 'year', 'month', 'day', 'hour', 'minute', 'second', 'note1', 'note2', 'storage', 'varName'],
```
Comment (DetroitWhiskey136) [inline actions/create_embed_message.js:15]: undo this change please

File: actions/create_embed_message.js [hunk 2]
```diff
@@ -181,7 +245,8 @@ module.exports = {
           timestamp2.style.display = 'none'
           note.style.display = 'none'
           note2.style.display = 'none'
-          authorUrl.placeholder = 'Unavaible!'
+          authorUrl.placeholder = 'Unavailable!'
```
Comment (DetroitWhiskey136) [inline actions/create_embed_message.js:248]: lol glad someone fixed that

File: actions/create_embed_message.js [hunk 3]
```diff
@@ -227,7 +292,11 @@ module.exports = {
       }
       // Author Name
       if (data.author) {
-        embed.setAuthor(this.evalMessage(data.author, cache), this.evalMessage(data.authorIcon, cache), this.evalMessage(data.authorUrl, cache))
```
Comment (DetroitWhiskey136) [inline actions/create_embed_message.js:230]: adding new lines just grows the file size try to minimize that.

File: actions/create_embed_message.js [hunk 4]
```diff
@@ -274,7 +359,10 @@ module.exports = {
         embed.setURL(this.evalMessage(data.url, cache))
       }
       if (data.author && data.authorIcon) {
-        embed.setAuthor(this.evalMessage(data.author, cache), this.evalMessage(data.authorIcon, cache))
```
Comment (DetroitWhiskey136) [inline actions/create_embed_message.js:277]: and here

File: actions/create_embed_message.js [hunk 5]
```diff
@@ -253,11 +327,22 @@ module.exports = {
...
           } else {
             embed.setTimestamp()
-            console.log('Invaild UTC timestamp! Changed from [String Timestamp] to [Current Timestamp].')
+            console.log(
+              'Invaild UTC timestamp! Changed from [String Timestamp] to [Current Timestamp].'
+            )
           }
           break
         case 'custom':
-          embed.setTimestamp(new Date(year || null, month || null, day || null, hour || null, minute || null, second || null))
```
Comment (DetroitWhiskey136) [inline actions/create_embed_message.js:260]: and here

File: actions/create_embed_message.js [hunk 6]
```diff
@@ -253,11 +327,22 @@ module.exports = {
             embed.setTimestamp(new Date(`${text}`))
           } else {
             embed.setTimestamp()
-            console.log('Invaild UTC timestamp! Changed from [String Timestamp] to [Current Timestamp].')
```
Comment (DetroitWhiskey136) [inline actions/create_embed_message.js:256]: same here

File: actions/create_embed_message.js (hunk 1 above)
Comment (DetroitWhiskey136) [reply actions/create_embed_message.js:15]: if the line length is to long just split it into two lines

**Review by DetroitWhiskey136 — APPROVED** [review]
LGTM

**Comment by DetroitWhiskey136** [comment]
Almeida made a good point, what if people want to set the description but not create a new embed each time, like when your edit an embed, this would be probably with this change

**Comment by Michael55555** [comment]
Yes, this is indeed a good point! But that's also the reason for not deleting the "set embed description" action, so that people still have the same options. Anyways, it would be great to make people edit their embeds with edit embed. We can release options from time to time that make this shift happen to the point where people create embeds with "Create Embed Message" and edit them with "Edit Embed Message".

**Comment by almeidx** [comment]
If we're going to add the description part to Create Embed Message, we might as well add Fields and Footer support

**Comment by Michael55555** [comment]
I'm working on that at the moment

**Comment by Michael55555** [comment]
Unfortunately I don't have time to add the fields atm. I would be glad if someone else takes this over.

**Comment by Michael55555** [comment]
> Unnecessary. This would defeat the purpose of the `Set Embed Description` action.

Alright agreed

**Pull request closed without merging**